### ISSUE TITLE
Impossible to send a message with mentions

### ISSUE THREAD

**Issue by ldavin** [GetStream/stream-chat-android-client#44]
**Describe the bug**
Hi, I've found it's currently impossible to add mentions to messages.
The `mentionedUsers` field of the Message model is annotated with `@IgnoreSerialisation` and does not appear in the body of the HTTP POST call (that we can see with the ChatLogLevel.ALL).

**SDK version**
- 1.10.0

**To Reproduce**
```kotlin
channel.sendMessage(
    Message(
        text = "stuff",
        mentionedUsers = mutableListOf(User("id"))
    )
).execute()
```

the resulting SDK call in the logs is:
```
I/Chat:Http: --> POST https://chat-us-east-1.stream-io-api.com/channels/messaging/XXX/message?api_key=XXX&user_id=XXXclient_id=XXX
I/Chat:Http: Content-Length: 331
I/Chat:Http: Content-Type: application/json
I/Chat:Http: stream-auth-type: jwt
I/Chat:Http: Accept-Encoding: application/gzip
I/Chat:Http: X-STREAM-CLIENT: 1.10.0-release
I/Chat:Http: Cache-Control: no-cache
I/Chat:Http: {"message":{"date":"","silent":false,"attachments":[],"show_in_channel":false,"isToday":false,"isStartDay":false,"isYesterday":false,"html":"","commandInfo":{},"id":"XXX","text":"XXX","time":"","cid":""}}
I/Chat:Http: --> END POST (331-byte body)
```

The mentioned users is not in the payload.

**Expected behavior**
The mentionedUsers should be sent to the server somehow 😅 

**Device:**
Simulator & OnePlus 6

Would it be possible to release a hotfixed version of the SDK? It's quite inconvenient.
Thanks!

**Comment by elevenetc** [comment]
Hello @ldavin thanks for report, that's indeed a bug. Although removing `IgnoreSerialisation` doesn't solve the problem as API doesn't accept objects. I'm in contact with backend team and keep you posted.

**Comment by elevenetc** [comment]
Fix is released in `1.11.0`.
To send mentioned users you need to user id in `mentionedUsersIds`:
```kotlin
val message = Message(text = "hello")
message.mentionedUsersIds.add("user-id")
client.sendMessage(channelType, channelId, message).enqueue { result ->
  val message = result.data()
}
```
In response mentioned user objects will be in field `mentionedUsers`:
```kotlin
val message = result.data()
val mentionedUsers = message.mentionedUsers
```

Looks a bit tricky. That's because backend has the same field name(`mentioned_users`) for incoming and outgoing message, but with different types: outgoing message needs to contains array of string ids, and incoming array of user objects.
